internal/generator/vector/filter/drop/filter.go

```diff
@@ -21,15 +21,15 @@ func (f *Filter) VRL() (string, error) {
 		condList := []string{}
 		for _, cond := range test.DropConditions {
 			if cond.Matches != "" {
-				condList = append(condList, fmt.Sprintf(`match(%s, r'%s')`, cond.Field, cond.Matches))
+				condList = append(condList, fmt.Sprintf(`match(to_string!(%s), r'%s')`, cond.Field, cond.Matches))
```

Comment:
Do we have a test to support this? Maybe one for the 403 in the bug

Reply:
https://github.com/openshift/cluster-logging-operator/pull/2749/files#diff-6282421d5e27209320b53a3de7f8c3c2885b46e9ae27b608b2fb52786decfdecR85